app/src/main/java/com/example/habitsmasher/ui/history/HabitEventListFragment.java

```diff
@@ -47,54 +52,61 @@
-        HabitEventListFragment fragment = new HabitEventListFragment(parentHabit, parentUser);
-        return fragment;
-    }
-
-    @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
+    public HabitEventListFragment () {
+        // Required empty constructor
     }
 
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-        // Get context and query
+        if (getArguments() != null) {
+            _parentHabit = (Habit) getArguments().getSerializable("parentHabit");
+            _username = (String) getArguments().getSerializable("parentUser");
+        }
+
+        _parentHabit.setHabitEvents(new HabitEventList());
+        _habitEventList = _parentHabit.getHabitEvents();
+
+        // Get context
         Context context = getContext();
-        Query query = getListOfHabitEventsFromFirebase(_username);
 
         try {
+            // Get query
+            Query query = getListOfHabitEventsFromFirebase(_username);
+
             // Populate the list with existing items in the database
             FirestoreRecyclerOptions<HabitEvent> options = new FirestoreRecyclerOptions.Builder<HabitEvent>()
                     .setQuery(query, HabitEvent.class)
                     .build();
 
+
+            Task<QuerySnapshot> querySnapshotTask = query.get();
+            /*
+            populate HabitList with current Habits and habit IDs to initialize state to match
+            database, fills when habitList is empty and snapshot is not, which is only
+            when app is initially launched
+            */
+            if (_habitEventList.getHabitEvents().isEmpty()) {
+                while (!querySnapshotTask.isComplete());
+                List<DocumentSnapshot> snapshotList = querySnapshotTask.getResult().getDocuments();
+                for (int i = 0; i < snapshotList.size(); i++) {
+                    Map<String, Object> extractMap = snapshotList.get(i).getData();
+                    String comment = (String) extractMap.get("comment");
+                    Timestamp date = (Timestamp) extractMap.get("date");
+                    String id = extractMap.get("id").toString();
+                    HabitEvent addHabitEvent = new HabitEvent(date.toDate(), comment, id);
+                    Log.d(TAG, addHabitEvent.getId());
+                    _habitEventList.addHabitEventLocally(addHabitEvent);
+                }
+            }
+
             // Set item adapter and habit event list
-            _habitEventItemAdapter = new HabitEventItemAdapter(options);
-            _habitEventList = _parentHabit.getHabitEvents();
+            _habitEventItemAdapter = new HabitEventItemAdapter(options, _parentHabit, _username, _habitEventList, _fragment);
         }
-        catch (Error e){
+        catch (NullPointerException e){
```

Comment:
If there's no events, does the list just default to empty locally?

Reply:
Yes! We set it as an empty list at the start